### DISCUSSION TITLE
Double substitution on Github Action - Nested substitution

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [community/community#26419] [Actions]
<p>I have a problem with a double substitution on a GitHub Action. I have a secret named: <code>MASTER_ENVIRONMENT_NAME</code> and <code>STAGING_ENVIRONMENT_NAME</code></p>
<p>On GitHub actions, I am trying to use the value of the env variable depending on the branch like this:</p>
<pre><code class="lang-auto">BRANCH_NAME=${GITHUB_REF##*/}
DEPLOYMENT=secrets.${BRANCH_NAME^^}_ENVIRONMENT_NAME
echo ${{ secrets.${BRANCH_NAME^^}_ENVIRONMENT_NAME }} // Failing here
</code></pre>
<p>I have tried multiple combinations but all failing, any thought?</p>

**Comment by Simran-B** [comment]
<p>Maybe like this:</p>
<pre><code class="lang-yaml">- name: Determine branch name
  id: branch-name
  run: |
    BRANCH_NAME="${GITHUB_REF##*/}"
    echo "::set-output name=upper::${BRANCH_NAME^^}"
- name: Pick secret dynamically
  run: |
    echo "secrets.${{ steps.branch-name.outputs.upper }}_ENVIRONMENT_NAME:"
    echo "${{ secrets[format('{0}_ENVIRONMENT_NAME', steps.branch-name.outputs.upper)] }}"
</code></pre>

**Comment by juancarlosjr97** [comment, marked as answer]
<p>Hi <a class="mention" href="/simran-b">@Simran-B</a> thank you for your reply.</p>
<p>Actually, I found the solution not that long ago doing the following:</p>
<pre><code class="lang-auto">- name: Set Branch Credentials
  run: |
    BRANCH_NAME=${GITHUB_REF##*/}
    BRANCH_NAME=${BRANCH_NAME^^}
    echo "BRANCH_ENVIRONMENT_NAME=${BRANCH_NAME}_ENVIRONMENT_NAME" &gt;&gt; $GITHUB_ENV
    
- name: Deploy 
  run: echo ${{ secrets[env.BRANCH_ENVIRONMENT_NAME] }}
</code></pre>
<p>Thank you very much!</p>

**Comment by Simran-B** [comment]
<p>Cool, I think yours is easier to parse.</p>
<p>BTW. Is there a way to combine <code>${NAME##...}</code> and <code>${NAME^^}</code> in bash?</p>
<p><strong>//edit:</strong> Bash doesn’t seem to support nesting<br>
</p><aside class="onebox stackexchange">
  <header class="source">
      <a href="https://stackoverflow.com/questions/48430580/bash-nested-substring-removal-extraction" target="_blank" rel="noopener">stackoverflow.com</a>
  </header>
  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/6231220/chris" target="_blank" rel="noopener">
    <img alt="Chris" src="https://user-images.githubusercontent.com/7819991/181097582-bac8fda1-6a6e-4a15-9c45-99269b479ac0.png" class="thumbnail onebox-avatar" width="128" height="128">
  </a>
<h4>
  <a href="https://stackoverflow.com/questions/48430580/bash-nested-substring-removal-extraction" target="_blank" rel="noopener">Bash Nested Substring Removal (Extraction)?</a>
</h4>

<div class="tags">
  <strong>string, bash, substring</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/6231220/chris" target="_blank" rel="noopener">
    Chris
  </a>
  on <a href="https://stackoverflow.com/questions/48430580/bash-nested-substring-removal-extraction" target="_blank" rel="noopener">07:52PM - 24 Jan 18 UTC</a>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by juancarlosjr97** [comment]
<p>I am not sure if it is possible to combine <code>${NAME##...}</code> and <code>${NAME^^}</code></p>
<p>However, it is possible to do this</p>
<pre><code class="lang-auto">TEST="part-1/part-2/part-needed"
echo ${TEST##*/} | tr [a-z] [A-Z] // PART-NEEDED
</code></pre>
<p>Thank you :slight_smile:</p>
